renderdoc/driver/shaders/dxbc/dxbc_debug.h

```diff
@@ -147,6 +147,55 @@ struct GlobalState
   std::map<SampleEvalCacheKey, ShaderVariable> sampleEvalCache;
 };
 
+struct SampleGatherResourceData
```

Comment:
It probably makes sense to put those structs inside `class DebugAPIWrapper` so they don't occupy global namespace?

Reply:
I did that initially, but the full name then felt very long, and I didn't want to remove 'SampleGather' from the struct name as that may cause someone to incorrectly assume that they have broader utility.

Reply:
Yes, also it's already in `namespace ShaderDebug`. Even if it were in the global namespace I don't think that's a big deal.